components/camel-websocket-jsr356/src/main/java/org/apache/camel/jsr356/JSR356WebSocketComponent.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.jsr356;
+
+import static java.util.Optional.ofNullable;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.websocket.Session;
+import javax.websocket.server.ServerContainer;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.impl.DefaultComponent;
+import org.apache.camel.spi.Metadata;
+
+public class JSR356WebSocketComponent extends DefaultComponent {
+    // didn't find a better way to handle that unless we can assume the CamelContext is in the ServletContext
+    private static final Map<String, ContextBag> SERVER_CONTAINERS = new ConcurrentHashMap<>();
+
+    @Metadata(label = "sessionCount")
+    protected int sessionCount;
+
+    @Override
+    protected Endpoint createEndpoint(final String uri, final String remaining, final Map<String, Object> parameters) {
+        return new JSR356Endpoint(this, uri);
+    }
+
+    public void sendMessage(final Session session, final Object message) throws IOException {
+        synchronized (session) {
+            // todo: handle async?
+            if (String.class.isInstance(message)) {
+                session.getBasicRemote().sendText(String.valueOf(message));
+            } else if (ByteBuffer.class.isInstance(message)) {
+                session.getBasicRemote().sendBinary(ByteBuffer.class.cast(message));
+            } else if (InputStream.class.isInstance(message)) {
+                int read;
+                final InputStream in = InputStream.class.cast(message);
+                final byte[] buffer = new byte[8192]; // todo: config
+                final OutputStream out = session.getBasicRemote().getSendStream();
+                while ((read = in.read(buffer)) >= 0) {
+                    out.write(buffer, 0, read);
+                }
```

Comment:
I guess you can reuse some utilities from camel's org.apache.camel.util.IOHelper